proposals/0035-linalg-matrix.md

```diff
@@ -39,126 +39,106 @@ Matrix Operations](/proposals/0031-hlsl-vector-matrix-operations.md).
+                     bool ColMajor, uint Align = sizeof(ElementType));
 
   template <typename T>
-    requires ArithmeticScalar<T>
-  static Matrix Load(groupshared T Arr[], uint StartIdx, uint Stride,
-                     bool ColMajor);
+  static typename hlsl::enable_if<hlsl::is_arithmetic<T>::value, Matrix>::type
+  Load(/*groupshared*/ T Arr[], uint StartIdx, uint Stride, bool ColMajor);
 
-  void
-  Store(RWByteAddressBuffer Res, uint StartOffset, uint Stride, bool ColMajor,
-        uint Align = sizeof(ElementType));
+  void Store(RWByteAddressBuffer Res, uint StartOffset, uint Stride,
+             bool ColMajor, uint Align = sizeof(ElementType));
 
   template <typename T>
-    requires ArithmeticScalar<T>
-  void Store(groupshared T Arr[], uint StartIdx, uint Stride,
-             bool ColMajor);
+  typename hlsl::enable_if<hlsl::is_arithmetic<T>::value, void>::type
+  Store(/*groupshared*/ T Arr[], uint StartIdx, uint Stride, bool ColMajor);
 
   // Row accesses
   vector<ElementType, M> GetRow(uint Index);
   void SetRow(vector<ElementType, M> V, uint Index);
 
   // Element access
-  std::enable_if_t<__detail::ComponentTypeTraits<ComponentTy>::IsNativeScalar,
-                   ElementType>
+  typename hlsl::enable_if<
+      __detail::ComponentTypeTraits<ComponentTy>::IsNativeScalar,
+      ElementType>::type
   Get(uint2 Index);
-  std::enable_if_t<__detail::ComponentTypeTraits<ComponentTy>::IsNativeScalar,
-                   void>
+  typename hlsl::enable_if<
+      __detail::ComponentTypeTraits<ComponentTy>::IsNativeScalar, void>::type
   Set(ElementType V, uint2 Index);
 
-  template <MatrixComponentType LHSTy, MatrixComponentType RHSTy, unit K>
-  std::enable_if_t<Use == MatrixUse::Accumulator && Scope == MatrixScope::Wave,
-                   void>
+  template <MatrixComponentType LHSTy, MatrixComponentType RHSTy, uint K>
+  typename hlsl::enable_if<
+      Use == MatrixUse::Accumulator && Scope == MatrixScope::Wave, void>::type
   MultiplyAccumulate(const Matrix<LHSTy, M, K, MatrixUse::A, Scope>,
                      const Matrix<RHSTy, K, N, MatrixUse::B, Scope>);
 
-  template <MatrixComponentType LHSTy, MatrixComponentType RHSTy, unit K>
-  std::enable_if_t<Use == MatrixUse::Accumulator && Scope == MatrixScope::Wave,
-                   void>
+  template <MatrixComponentType LHSTy, MatrixComponentType RHSTy, uint K>
+  typename hlsl::enable_if<
+      Use == MatrixUse::Accumulator && Scope == MatrixScope::Wave, void>::type
   SumAccumulate(const Matrix<LHSTy, M, K, MatrixUse::A, Scope>,
                 const Matrix<RHSTy, K, N, MatrixUse::B, Scope>);
 
   // Cooperative Vector outer product accumulate.
   template <typename T>
-  std::enable_if_t<Use == MatrixUse::Accumulator, void>
-  OuterProductAccumulate(const vector<T, M> &, const vector<T, N> &);
+  typename hlsl::enable_if<Use == MatrixUse::Accumulator, void>::type
+  OuterProductAccumulate(const vector<T, M>, const vector<T, N>);

```

Comment:
would a comment like /* inout */ be useful here to act as a counterpart to &?

Reply:
I mostly decided that we didn't actually need the reference since this will just be inlined it won't actually create a copy.